content/protocols/mqtt.textile

```diff
@@ -0,0 +1,150 @@
+---
+title: MQTT
+meta_description: "Any MQTT-enabled client can communication with the Ably service through the Ably MQTT protocol adapter. This is especially useful where an Ably client library SDK is not available for your language of choice."
+---
+
+The Ably MQTT protocol adapter is able to translate back and forth between "MQTT":https://mqtt.org/ and Ably's own protocol, allowing for seamless integration of any systems you may have. MQTT (MQ Telemetry Transport) is a "publish/subscribe":https://ably.com/topic/pub-sub , lightweight messaging protocol designed for constrained devices and low-bandwidth networks. One of the major uses of MQTT is with IoT (Internet of Things), where these principles are key to having effective communication between various devices.
+
+h2(#when-to-use). When to use the MQTT adapter
+
+The MQTT adapter is our recommended way of interacting with Ably from devices which do not have a native Ably client library, such as Arduino platforms, C/C++ applications, and so on. Anyone who has previously been using Pubnub or Pusher client libraries for this purpose may want to consider switching to MQTT. Compared to the Pubnub protocol, using MQTT will result in better performance with lower latency. Compared to the Pusher protocol, MQTT will give you connection state recovery.
+
+We recommend you use MQTT to interact with Ably when:
+
+* Using devices which do not have a device-specific Ably Client Library SDK, such as Arduino platforms, C/C++ applications, and so on
+* If bandwidth is limited and you want to keep network traffic to a minimum
+* You want to keep network traffic to a minimum.
+
+Note, we do not currently support MQTT over websockets but let us know if you are interested.
+
+h2(#limitations). Known limitations 
+
+Our MQTT adapter only supports features supported by both the MQTT protocol and the Ably platform. So:
+
+* It only supports MQTT 3.1.1 clients only. Connection attempts using earlier protocol versions will be rejected
+* Publishing supports QoS 0 or 1. Subscribing only supports QoS 0
+* Session resumption is supported within the usual "Ably time limit of two minutes":/history-storage/storage
+* It doesn’t support any MQTT features that aren’t normally supported by Ably, such as @WILL@ messages, the @RETAIN@ flag or "wildcard channel subscriptions":/channels
+* It doesn't support any Ably features that aren't supported by the MQTT protocol, such as presence, history and push notifications (though presence may be supported soon). Of course, you can always use the Ably REST API in conjunction with the MQTT adapter to access features available over REST
+
+h2(#config). Configuration
+
+We recommend that you always make use of "Token Authentication over Basic Authentication":/auth when trying to connect from devices you may not trust. In addition, if you're using Basic Authentication you'll be required to use SSL to connect to our adapter to ensure the API key cannot be accessed by someone listening in.
```

Comment:
This section would be better served under "authentication" I think. Also don't need to capitalise token and basic auth.